plugins/transformations/json/srcC/FlatTypeTransformation.c

```diff
@@ -649,22 +760,26 @@ DDS_Boolean RTI_TSFM_Json_FlatTypeTransformation_add_eos_octet_seq(
     max = DDS_OctetSeq_get_maximum(self);
     length = DDS_OctetSeq_get_length(self);
 
-    /* The sequence is well terminated, no need to add EOS */
-    if (DDS_OctetSeq_get_reference(self, length - 1) == '\0') {
+    /* The sequence is well terminated, no need to add a nul terminator */
+    if (*DDS_OctetSeq_get_reference(self, length - 1) == '\0') {
         return DDS_BOOLEAN_TRUE;
     }
 
-    /* The EOS character cannot be added without allocating new memory */
+    /* The nul terminator cannot be added without allocating new memory */
     if (max == length) {
-        return DDS_BOOLEAN_FALSE;
-    }
-
-    /*
-     * In this case, we have at least one 'emtpy' space in the sequence,
-     * therefore, we can add the EOS character.
-     */
-    if (!DDS_OctetSeq_set_length(self, length + 1)) {
-        return DDS_BOOLEAN_FALSE;
+        if (!DDS_OctetSeq_ensure_length(self, length + 1, length + 1)) {
+            RTI_TSFM_ERROR("failed to ensure_length of an octet sequence")
+            return DDS_BOOLEAN_FALSE;
+        }
+    } else {
+        /*
+         * In this case, we have at least one emtpy space in the sequence,
+         * therefore, we can add the nul terminator.
+         */
+        if (!DDS_OctetSeq_set_length(self, length + 1)) {
+            RTI_TSFM_ERROR("failed to set_length of an octet sequence")
+            return DDS_BOOLEAN_FALSE;
+        }
     }
```

Comment:
This whole block is equivalent to just calling `DDS_OctetSeq_ensure_length(self, length + 1, length + 1)`, since `ensure_length()` should not reallocate the sequence if the max is less than the current.

Reply:
good point